debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -397,6 +412,34 @@ public JdbcConnection query(String query, StatementFactory statementFactory, Res
         return this;
     }
 
+    /**
+     * Execute a SQL query and map the result set into an expected type.
+     * @param <T> type returned by the mapper
+     *
+     * @param query the SQL query
+     * @param statementFactory the function that should be used to create the statement from the connection; may not be null
+     * @param mapper the function processing the query results
+     * @return the result of the mapper calculation
+     * @throws SQLException if there is an error connecting to the database or executing the statements
+     * @see #execute(Operations)
+     */
+    public <T> T queryAndMap(String query, StatementFactory statementFactory, ResultSetMapper<T> mapper) throws SQLException {
+        Connection conn = connection();
+        try (Statement statement = statementFactory.createStatement(conn);) {
+            if (LOGGER.isTraceEnabled()) {
+                LOGGER.trace("running '{}'", query);
+            }
+            try (ResultSet resultSet = statement.executeQuery(query);) {
+                if (mapper != null) {
```

Comment:
Let's pull the parameter check to the beginning of the method:

    Objects.requireNotNull(mapper);